Nos/Views/LinkPreview.swift

```diff
@@ -1,47 +1,63 @@
+import Dependencies
 import SwiftUI
 import LinkPresentation
 import SDWebImageSwiftUI
 
 struct ImageLinkButton: View {
     let url: URL
     
+    @State private var isViewerPresented: Bool = false
+
     var body: some View {
-        Button(action: {
-            UIApplication.shared.open(url)
-        }, label: {
-            let image = WebImage(url: url)
-                .resizable()
-                .indicator(.activity)
-            
-            ZStack {
-                image
-                    .blur(radius: 10)
-                    .clipped()
-                
-                image
-                    .aspectRatio(contentMode: .fit)
-                    .frame(maxWidth: .infinity, maxHeight: .infinity)
+        Button(
```

Comment:
Given that Button is a well-known component in SwiftUI, I prefer the more succinct declaration:

```
Button {
    isViewerPresented 0 true
} label: {
    ....
}
```